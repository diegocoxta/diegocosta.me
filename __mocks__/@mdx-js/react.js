module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
