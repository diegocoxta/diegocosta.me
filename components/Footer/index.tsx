import styles from './styles.module.css';

interface FooterProps {
  sourceCode?: string;
  author: string;
}

export default function Footer(props: FooterProps): React.ReactElement {
  const year = new Date().getFullYear();

  return (
    <footer className={styles.container}>
      <p className={styles.label}>
        CC-BY {year} <span>{props.author}</span> â€¢{' '}
        {props.sourceCode && (
          <a
            className={styles.link}
            href={props.sourceCode}
            data-testid="footer-source-code"
            target="__blank"
            rel="noopener"
          >
            source code
          </a>
        )}
      </p>
    </footer>
  );
}
