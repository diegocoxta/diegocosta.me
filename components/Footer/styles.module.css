.container {
  composes: container from '~/components/Container/styles.module.css';
  margin-bottom: 50px;

  nav {
    margin-bottom: 50px;
  }

  @media (width >= 760px) {
    nav {
      margin-bottom: 0;
    }
  }
}

.label {
  color: var(--main-text-color);
  line-height: 1.5;
  margin: 0;
  padding: 0;
  text-align: center;
}

.link {
  color: var(--main-accent-color);
  text-decoration: none;
  box-shadow: none;

  &:hover,
  &:focus {
    border-bottom: 1px solid var(--main-accent-color);
    outline: none;
  }
}
