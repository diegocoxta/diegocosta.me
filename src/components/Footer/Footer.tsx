import React from 'react';
import styled from 'styled-components';

import { useLocale } from '~/hooks/useLocale';

const Content = styled.footer`
  margin-bottom: 50px;

  nav {
    margin-bottom: 50px;
  }

  @media (min-width: 760px) {
    nav {
      margin-bottom: 0;
    }
  }
`;

const Label = styled.span`
  color: ${({ theme }) => theme.textColor};
  line-height: 1.5;
`;

const Link = styled.a.attrs({
  target: '_blank',
  rel: 'noopener',
})`
  color: ${({ theme }) => theme.accentColor};
  text-decoration: none;
  box-shadow: none;

  :hover,
  :focus {
    border-bottom: 1px solid ${({ theme }) => theme.accentColor};
    outline: none;
  }
`;

interface FooterProps {
  year: number;
  sourceCode?: string;
}

export default function Footer(props: FooterProps): React.ReactElement {
  const locale = useLocale();

  return (
    <>
      <Content>
        <Label>
          CC-BY {props.year}, {locale.getTranslationFor('footer.builtWith')}{' '}
        </Label>
        <Link href="https://gatsbyjs.org">gatsby</Link>
        <Label> â€¢ </Label>
        {props.sourceCode && <Link href={props.sourceCode}>{locale.getTranslationFor('footer.sourceCode')}</Link>}
      </Content>
    </>
  );
}
