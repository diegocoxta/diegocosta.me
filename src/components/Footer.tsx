import React from 'react';
import styled from 'styled-components';

import { useLocale } from '@app/components/LanguageSwitcher';

const Container = styled.footer`
  margin-bottom: 50px;

  nav {
    margin-bottom: 50px;
  }

  @media (min-width: 760px) {
    nav {
      margin-bottom: 0;
    }
  }
`;

const Label = styled.span`
  color: ${({ theme }) => theme.textColor};
  line-height: 1.5;
`;

const Link = styled.a.attrs({
  target: '_blank',
  rel: 'noopener',
})`
  color: ${({ theme }) => theme.accentColor};
  text-decoration: none;
  box-shadow: none;

  &:hover,
  &:focus {
    border-bottom: 1px solid ${({ theme }) => theme.accentColor};
    outline: none;
  }
`;

interface FooterProps {
  sourceCode?: string;
  author?: string;
}

export default function Footer(props: FooterProps): React.ReactElement {
  const locale = useLocale();
  const year = new Date().getFullYear();

  return (
    <Container>
      <Label>
        CC-BY {year} {props.author}, {locale.getTranslationFor('footer.builtWith')}{' '}
        <Link href="https://gatsbyjs.org">gatsby</Link> â€¢{' '}
        {props.sourceCode && <Link href={props.sourceCode}>{locale.getTranslationFor('footer.sourceCode')}</Link>}
      </Label>
    </Container>
  );
}
