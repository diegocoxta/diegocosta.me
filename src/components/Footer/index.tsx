import React from 'react';
import { useStaticQuery, graphql } from 'gatsby';

import Footer from './Footer';

export default (): React.ReactElement => {
  const {
    site: {
      siteMetadata: { sourceCode },
    },
  } = useStaticQuery(
    graphql`
      query FooterComponent {
        site {
          siteMetadata {
            sourceCode
          }
        }
      }
    `
  );

  const year = new Date().getFullYear();

  return <Footer sourceCode={sourceCode} year={year} />;
};
