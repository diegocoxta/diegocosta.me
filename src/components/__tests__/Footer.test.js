import React from 'react';
import { render } from '@testing-library/react';

import Footer from '../Footer';

describe('<Footer />', () => {
  it('renders properly', () => {
    const { baseElement, queryByText } = render(<Footer year={2020} />);
    expect(queryByText('footer.sourceCode')).toBeFalsy();
    expect(baseElement).toMatchSnapshot();
  });

  it('renders the sourceCode url if defined', () => {
    const { baseElement, getByText } = render(
      <Footer year={2020} sourceCode="https://github.com/diegocoxta/diegocosta.me" />
    );
    expect(getByText('footer.sourceCode')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
