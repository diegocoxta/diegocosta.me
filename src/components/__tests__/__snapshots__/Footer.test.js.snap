// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> do not render the sourceCode url and author if not defined 1`] = `
.c0 {
  max-width: 960px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  margin-bottom: 50px;
}

.c1 nav {
  margin-bottom: 50px;
}

.c2 {
  line-height: 1.5;
  margin: 0;
  padding: 0;
  text-align: center;
}

.c3 {
  text-decoration: none;
  box-shadow: none;
}

.c3:hover,
.c3:focus {
  border-bottom: 1px solid;
  outline: none;
}

@media (min-width: 760px) {
  .c1 nav {
    margin-bottom: 0;
  }
}

<body>
  <div>
    <section
      class="c0 c1"
    >
      <p
        class="c2"
      >
        CC-BY 
        2025
         
        <span />
        , built with 
        <a
          class="c3"
          href="https://gatsbyjs.org"
          rel="noopener"
          target="_blank"
        >
          gatsby
        </a>
         •
         
      </p>
    </section>
  </div>
</body>
`;

exports[`<Footer /> renders properly 1`] = `
.c0 {
  max-width: 960px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  margin-bottom: 50px;
}

.c1 nav {
  margin-bottom: 50px;
}

.c2 {
  line-height: 1.5;
  margin: 0;
  padding: 0;
  text-align: center;
}

.c3 {
  text-decoration: none;
  box-shadow: none;
}

.c3:hover,
.c3:focus {
  border-bottom: 1px solid;
  outline: none;
}

@media (min-width: 760px) {
  .c1 nav {
    margin-bottom: 0;
  }
}

<body>
  <div>
    <section
      class="c0 c1"
    >
      <p
        class="c2"
      >
        CC-BY 
        2025
         
        <span>
          diegocoxta
        </span>
        , built with 
        <a
          class="c3"
          href="https://gatsbyjs.org"
          rel="noopener"
          target="_blank"
        >
          gatsby
        </a>
         •
         
      </p>
    </section>
  </div>
</body>
`;

exports[`<Footer /> renders the sourceCode url if defined 1`] = `
.c0 {
  max-width: 960px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  margin-bottom: 50px;
}

.c1 nav {
  margin-bottom: 50px;
}

.c2 {
  line-height: 1.5;
  margin: 0;
  padding: 0;
  text-align: center;
}

.c3 {
  text-decoration: none;
  box-shadow: none;
}

.c3:hover,
.c3:focus {
  border-bottom: 1px solid;
  outline: none;
}

@media (min-width: 760px) {
  .c1 nav {
    margin-bottom: 0;
  }
}

<body>
  <div>
    <section
      class="c0 c1"
    >
      <p
        class="c2"
      >
        CC-BY 
        2025
         
        <span>
          @diegocoxta
        </span>
        , built with 
        <a
          class="c3"
          href="https://gatsbyjs.org"
          rel="noopener"
          target="_blank"
        >
          gatsby
        </a>
         •
         
        <a
          class="c3"
          data-testid="footer-source-code"
          href="https://github.com/diegocoxta/diegocosta.me"
          rel="noopener"
          target="_blank"
        >
          source code
        </a>
      </p>
    </section>
  </div>
</body>
`;
